qiskit/circuit/classical/types/__init__.py

```diff
@@ -40,6 +40,10 @@
 .. autoclass:: Bool
 .. autoclass:: Uint
 
+All types have a :attr:`~Type.const` field to indicate their const-ness. When the result type of
+an expression is constant, the expression is considered a constant expression. Constant
+expressions can be used in certain contexts that aren't valid for runtime-initialized variables.
```

Comment:
Thanks for the text.

Can you comment on briefly on what we mean by "const-ness", just to make clear we mean something more like the Rust `const` or C++ `constexpr` (can be compile-time evaluated) than the C `const` (pinky promise not to write to it at runtime through this identifier, unless I change my mind later)?

Reply:
Done in 166d7b2